src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -4,13 +4,16 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
+using System.Text;
 using Microsoft.Diagnostics.Tracing;
 
 namespace Microsoft.Diagnostics.Monitoring.EventPipe
 {
     internal static class TraceEventExtensions
     {
-        public static bool TryGetCounterPayload(this TraceEvent traceEvent, CounterFilter filter, string sessionId, out ICounterPayload payload)
+        private static IDictionary<string, bool> inactiveSharedSessions = new Dictionary<string, bool>();
```

Comment:
When a SHARED session's unique identifier is notified of a `MultipleSessionsConfiguredIncorrectlyError`, that unique identifier goes here - otherwise, it wouldn't know that it's not allowed to listen to the normal SHARED events coming down for existing sessions.

Reply:
It looks like we never use the bool, could this be a HashSet<string> instead?

Reply:
Agreed, HashSet initialized with an appropriate StringComparer looks like it should work here.